### DISCUSSION TITLE
How to implement comments and score

### DISCUSSION THREAD

**Discussion by profispojka** [pocketbase/pocketbase#695] [Q&A] [answered]
What would be the best way I can implement the ability for the user to comment and a star rating to a post?

Comment would just have a user name appended to it and a string value. How can I add multiple comments to a post? Should I just create a separate collection for coments and the filter a comments for specific post from that collection?

How to implement a rating? If multiple users rate a single post, I only need the final average of the rattings, can I get that right now, or do I have to get all the ratings filtered by post and then do the averaging on the FE?

Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
You can create 2 collections - `posts` and `comments` and for the `comments` collection you can define for example the following fields:

- message (`text`)
- rating (`select` with 1,2,3,4,etc. as options)
- post (`relation` to the `posts` collection)
- user (`relation` to the `profiles` collection or `user` type field; note that those 2 field types will be soon merged per #376)

In the future we'll have support for aggregated/computed read only fields (see #311), but for now you'll have to do the average rating on your own client-side.

**Comment by profispojka** [comment]
This is awesome answer, thank you! I have another small question regarding this.

How can I sort a post based for example on number of its comments? Right now, the only solution I can think of is loading all existing comments and, sort the data on FE and based on that I will get list of those posts like this: ?filter=(id='x' || id='y' || id = 'z')

But I dont think this solution will scale well. Is there currently a better one? Thank you very much.

**Reply by ganigeorgiev** [reply]
Sorry, server-side aggregations currently are supported only if you operate directly with the [database and the Dao service](https://pocketbase.io/docs/db-queries/) (more documentation on this will be available in the near future).

As mentioned in the previous comment, aggregated/computed read-only fields will be added as part of #311.

**Reply by profispojka** [reply]
Thanks for the answer Gani.
